### PR TITLE
[FIX] [ENHANCEMENT] logging records in notebook without ipython

### LINKED ISSUE #4973: [BUG] ModuleNotFoundError: No module named 'IPython'
**Describe the bug**
After installing the new SDK in a clean environment and start using the python console, the following error is raised:

```bash
  File "/Users/frascuchon/Projects/argilla/argilla/argilla/.venv/lib/python3.9/site-packages/argilla_sdk/__init__.py", line 15, in <module>
    from argilla_sdk.client import *  # noqa
  File "/Users/frascuchon/Projects/argilla/argilla/argilla/.venv/lib/python3.9/site-packages/argilla_sdk/client.py", line 23, in <module>
    from argilla_sdk._helpers._resource_repr import ResourceHTMLReprMixin
  File "/Users/frascuchon/Projects/argilla/argilla/argilla/.venv/lib/python3.9/site-packages/argilla_sdk/_helpers/_resource_repr.py", line 17, in <module>
    from IPython.display import HTML
ModuleNotFoundError: No module named 'IPython'
```

**To Reproduce**

**Expected behavior**

**Screenshots**

**Environment (please complete the following information):**
 - OS [e.g. iOS]:
 - Browser [e.g. chrome, safari]:
 - Argilla Version [e.g. 1.0.0]:
 - ElasticSearch Version [e.g. 7.10.2]:
 - Docker Image (optional) [e.g. argilla:v1.0.0]:

**Additional context**

### PR DESCRIPTION
**Pull request by burtenshaw**
This PR introduces these results in a notebook environment:

image

- tqdm shows a status bar
- non info level logging is printed with rich
- `DatasetRecords.log` returns self not `List[Record]` 
- html repr of resources is dealt with as string not ipython `HTML`

# Argilla Community Growers

Thanks for your contribution! As part of our Community Growers initiative 🌱, we're donating Justdiggit bunds in your name to reforest sub-Saharan Africa. To claim your Community Growers certificate, please contact David Berenstein in our Slack community or fill in this form https://tally.so/r/n9XrxK once your PR has been merged.

# Pull Request Templates

Please go the the `Preview` tab and select the appropriate sub-template:

* [🐞-bug](?expand=1&template=bug.md)
* [📚-documentation](?expand=1&template=docs.md)
* [🆕-features](?expand=1&template=features.md)

# Generic Pull Request Template

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #<issue_number>

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Review by frascuchon — CHANGES_REQUESTED** [review]
I think we should also remove refs to IPython in `client.py`

File: argilla-sdk/src/argilla_sdk/_helpers/_log.py [hunk 1]
```diff
@@ -36,10 +37,26 @@ def log_message(message: str, level: str = "info") -> None:
...
+
+
 class LoggingMixin:
     """A utility mixin for logging from a `Resource` class."""
 
     def _log_message(self, message: str, level: str = "info") -> None:
         class_name = self.__class__.__name__
         message = f"{class_name}: {message}"
-        log_message(level=level, message=message)
+        if self._is_interactive() and level != "info":
```
Comment (frascuchon) [inline argilla-sdk/src/argilla_sdk/_helpers/_log.py:54]: Why do we include the level != "info" as part of the condition? That means nothing will be logged when running into a notebook by default, right?

Comment (burtenshaw) [reply argilla-sdk/src/argilla_sdk/_helpers/_log.py:54]: Yeah, we only print error and warning in notebooks. Otherwise it becomes verbose.

**Review by frascuchon — APPROVED** [review]

File: argilla-sdk/src/argilla_sdk/records/_dataset_records.py [hunk 2]
```diff
@@ -237,7 +241,7 @@ def log(
             level="info",
         )
 
-        return created_or_updated
+        return self
```
Comment (frascuchon) [inline argilla-sdk/src/argilla_sdk/records/_dataset_records.py]: I think this change is making some integration tests fail. But maybe we can just this change here and create another PR to tackle https://github.com/argilla-io/argilla/issues/4947

**Review by frascuchon — APPROVED** [review]

**Review by davidberenstein1957 — APPROVED** [review]

**Pull request merged**